components/director/pkg/operation/directive.go

```diff
@@ -292,3 +299,16 @@ func executeSyncOperation(ctx context.Context, next gqlgen.Resolver, tx persiste
 func isErrored(app model.Entity) bool {
 	return (app.GetError() == nil || *app.GetError() == "")
 }
+
+func determineApplicationInProgressStatus(opType graphql.OperationType) model.ApplicationStatusCondition {
+	switch opType {
+	case graphql.OperationTypeCreate:
+		return model.ApplicationStatusConditionCreating
+	case graphql.OperationTypeUpdate:
+		return model.ApplicationStatusConditionUpdating
+	case graphql.OperationTypeDelete:
+		return model.ApplicationStatusConditionDeleting
+	default:
+		return model.ApplicationStatusConditionInitial
```

Comment:
Setting `Initial` as will lead to confusion and inconsitent state. We should rather return an error of some kind if opType is not among the supported ones.